[BUG] build error for `cpp/tests/io/parquet_test.cpp`

**Describe the bug**
```
/home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/tests/io/parquet_test.cpp: In member function ‘virtual void ParquetWriterTest_MultipleMismatchedSources_Test::TestBody()’:
/home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/tests/io/parquet_test.cpp:693:91: error: call of overloaded ‘source_info(<brace-enclosed initializer list>)’ is ambiguous
       cudf_io::parquet_reader_options::builder(cudf_io::source_info{{int5file, float5file}});
                                                                                           ^
In file included from /home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/include/cudf/io/parquet.hpp:19:0,
                 from /home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/tests/io/parquet_test.cpp:26:
/home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/include/cudf/io/types.hpp:167:12: note: candidate: cudf::io::source_info::source_info(const string&)
   explicit source_info(std::string const& file_path)
            ^~~~~~~~~~~
/home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/include/cudf/io/types.hpp:163:12: note: candidate: cudf::io::source_info::source_info(const std::vector<std::basic_string<char> >&)
   explicit source_info(std::vector<std::string> const& file_paths)
            ^~~~~~~~~~~
/home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/tests/io/parquet_test.cpp:699:90: error: call of overloaded ‘source_info(<brace-enclosed initializer list>)’ is ambiguous
       cudf_io::parquet_reader_options::builder(cudf_io::source_info{{int5file, int10file}});
                                                                                          ^
In file included from /home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/include/cudf/io/parquet.hpp:19:0,
                 from /home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/tests/io/parquet_test.cpp:26:
/home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/include/cudf/io/types.hpp:167:12: note: candidate: cudf::io::source_info::source_info(const string&)
   explicit source_info(std::string const& file_path)
            ^~~~~~~~~~~
/home/0.alpha/rapids.pkgs/20200917/cudf.dev.0.16.arena_memory_resource/cudf.dev/cpp/include/cudf/io/types.hpp:163:12: note: candidate: cudf::io::source_info::source_info(const std::vector<std::basic_string<char> >&)
   explicit source_info(std::vector<std::string> const& file_paths)
            ^~~~~~~~~~~
make[2]: *** [tests/CMakeFiles/PARQUET_TEST.dir/build.make:83: tests/CMakeFiles/PARQUET_TEST.dir/io/parquet_test.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1563: tests/CMakeFiles/PARQUET_TEST.dir/all] Error 2
make: *** [Makefile:161: all] Error 2

```

**Environment details**
commit 7e7b329857d5f849393982919ebfd1bc55601b4f
devtoolset-7
java/ci/build-in-docker.sh

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -135,6 +135,7 @@
 - PR #6304 Fix span_tests.cu includes
 - PR #6331 Avoids materializing `RangeIndex` during frame concatnation (when not needed)
 - PR #6278 Add filter tests for struct columns
+- PR #6345 Fix ambiguous constructor compile error with devtoolset
 
 
 # cuDF 0.15.0 (26 Aug 2020)

--- a/cpp/tests/io/parquet_test.cpp
+++ b/cpp/tests/io/parquet_test.cpp
@@ -689,14 +689,16 @@ TEST_F(ParquetWriterTest, MultipleMismatchedSources)
   auto const int5file = create_parquet_file<int>(5);
   {
     auto const float5file = create_parquet_file<float>(5);
+    std::vector<std::string> files{int5file, float5file};
     cudf_io::parquet_reader_options const read_opts =
-      cudf_io::parquet_reader_options::builder(cudf_io::source_info{{int5file, float5file}});
+      cudf_io::parquet_reader_options::builder(cudf_io::source_info{files});
     EXPECT_THROW(cudf_io::read_parquet(read_opts), cudf::logic_error);
   }
   {
     auto const int10file = create_parquet_file<int>(10);
+    std::vector<std::string> files{int5file, int10file};
     cudf_io::parquet_reader_options const read_opts =
-      cudf_io::parquet_reader_options::builder(cudf_io::source_info{{int5file, int10file}});
+      cudf_io::parquet_reader_options::builder(cudf_io::source_info{files});
     EXPECT_THROW(cudf_io::read_parquet(read_opts), cudf::logic_error);
   }
 }

